### ISSUE TITLE
What's the meaning of -100 in traing?

### ISSUE THREAD

**Issue by CheungZeeCn** [Shivanandroy/T5-Finetuning-PyTorch#3]
What's the meaning of -100 in  lm_labels[y[:, 1:] == tokenizer.pad_token_id] = -100

thank you~

```python 
def train(epoch, tokenizer, model, device, loader, optimizer):

    """
    Function to be called for training with the parameters passed from main function

    """

    model.train()
    for _, data in enumerate(loader, 0):
        y = data["target_ids"].to(device, dtype=torch.long)
        y_ids = y[:, :-1].contiguous()
        lm_labels = y[:, 1:].clone().detach()
        lm_labels[y[:, 1:] == tokenizer.pad_token_id] = -100
        ids = data["source_ids"].to(device, dtype=torch.long)
        mask = data["source_mask"].to(device, dtype=torch.long)

        outputs = model(
            input_ids=ids,
            attention_mask=mask,
            decoder_input_ids=y_ids,
            labels=lm_labels,
        )
        loss = outputs[0]

        if _ % 10 == 0:
            training_logger.add_row(str(epoch), str(_), str(loss))
            console.print(training_logger)

        optimizer.zero_grad()
        loss.backward()
        optimizer.step()

```

**Comment by pholur** [comment]
This is to convey to the loss function that uses conditional cross entropy at the token level, that this token classification result does not matter / can be ignored. This is important, since in this case, the -100 is applied to the target tokens that are the special `pad_token_id`. 
Declaration: I am not a co-author of this repo.

**Comment by CheungZeeCn** [comment]
Thx  a lot~
